### PR TITLE
Bread, Cake and Blood

### PR DESCRIPTION
**Pull request by pent0**

### CODE REVIEW

File: vita3k/ngs/include/ngs/ngs.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        std::vector<Block> blocks;      /// All block sorted by size
+
+        explicit SystemMemspaceBlockAllocator(const std::uint32_t memspace_size);
+        
+        std::uint32_t alloc(const std::uint32_t size);
+        bool free(const std::uint32_t offset);
+    };
+
+    struct System {
+        std::vector<Rack*> racks;
```
Comment (VelocityRa) [inline vita3k/ngs/include/ngs/ngs.h]: Any reason for using raw pointers?
```suggestion
        std::vector<std::unique_ptr<Rack>> racks;
```

File: vita3k/ngs/src/ngs.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        , alloc(memspace_size) {
+
+    }
+    
+    bool init(const MemState &mem, SystemInitParameters *parameters, Ptr<void> memspace, const std::uint32_t memspace_size) {
+        // Reserve first memory allocation for our System struct
+        System *sys = memspace.cast<System>().get(mem);
+        sys = new (sys) System(memspace, memspace_size);
+
+        sys->racks.resize(parameters->max_racks);
```
Comment (VelocityRa) [inline vita3k/ngs/src/ngs.cpp]: Change this too if you switch to `unique_ptr`.
```suggestion
        sys->racks.reserve(parameters->max_racks);
```

File: vita3k/ngs/include/ngs/ngs.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    };
+
+    struct System;
+
+    struct Voice {
+        std::uint32_t holdit;
+    };
+
+    struct Rack {
+        System *mama;
```
Comment (VelocityRa) [inline vita3k/ngs/include/ngs/ngs.h]: 👀

File: vita3k/ngs/include/ngs/ngs.h (hunk 1 above)
Comment (pent0) [reply vita3k/ngs/include/ngs/ngs.h]: Game allocs memory space for rack pointers.

**Comment by pent0** [comment]
Close due to lack of interest and knowledge

**Comment by VelocityRa** [comment]
@pent0 Isn't any of the work here mergeable?

**Comment by pent0** [comment]
Invalidate

**Pull request closed without merging**